src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -0,0 +1,378 @@
+        cbsd_records_iter_1 = {'registrationRequests': [device_1, device_2],
+                              'grantRequests': [grant_request_1, grant_request_2]}
+        cbsd_records_iter_2 = {'registrationRequests': [device_3, device_4],
+                              'grantRequests': [grant_request_3, grant_request_4]}
+        protected_entities_iter_1 = {'PPA': [ppa_record_1], 'ESC': [esc_record_1]}
+        protected_entities_iter_2 = {'GWPZ': [gwpz_record_1] , 'FSS': [fss_record_1]}
+
+
+        # SAS Test Harness FAD records
+        sas_th_cbsd_0 = json.load(
+          open(os.path.join('testcases', 'testdata', 'cbsd_0.json')))
+        sas_th_cbsd_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'cbsd_1.json')))
+        sas_th_cbsd_2 = json.load(
+          open(os.path.join('testcases', 'testdata', 'cbsd_2.json')))
+        sas_th_cbsd_3 = json.load(
+          open(os.path.join('testcases', 'testdata', 'cbsd_3.json')))
+        sas_th_cbsd_4 = json.load(
+          open(os.path.join('testcases', 'testdata', 'cbsd_4.json')))
+        sas_th_cbsd_5 = json.load(
+          open(os.path.join('testcases', 'testdata', 'cbsd_5.json')))
+
+        sas_th_0_cbsd_iter_0 = [sas_th_cbsd_0]
+        sas_th_0_cbsd_iter_1 = [sas_th_cbsd_1]
+        sas_th_0_cbsd_iter_2 = [sas_th_cbsd_2]
+        sas_th_1_cbsd_iter_0 = [sas_th_cbsd_3]
+        sas_th_1_cbsd_iter_1 = [sas_th_cbsd_4]
+        sas_th_1_cbsd_iter_2 = [sas_th_cbsd_5]
+    
+        # Load DPA record
+        dpa_0 = json.load(
+          open(os.path.join('testcases', 'testdata', 'dpa_0.json')))
+        dpa_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'dpa_1.json')))
+
+        sas_harness_0 = {
+                    'name': 'SAS-TH-1',
+                    'url': 'localhost:9001',
+                    'cert_file': 'certs/server_a.cert',
+                    'key_file': 'certs/server_a.key',
+                    'certificateHash': 'dummy',
+                    'cbsdRecord': [sas_th_0_cbsd_iter_0, sas_th_0_cbsd_iter_1, sas_th_0_cbsd_iter_2],
+                    'fileServePath': 'sas_harness_2/'
+                }
+        sas_harness_1 = {
+                    'name': 'SAS-TH-2',
+                    'url': 'localhost:9002',
+                    'cert_file': 'certs/server_b.cert',
+                    'key_file': 'certs/server_b.key',
+                    'certificateHash': 'dummy',
+                    'cbsdRecord': [sas_th_1_cbsd_iter_0, sas_th_1_cbsd_iter_1, sas_th_1_cbsd_iter_2],
+                    'fileServePath': 'sas_harness_1/'
+                }
+        # Create the actual config.
+        config = {
+            'cbsdRecordsN1': [cbsd_records_iter_1, cbsd_records_iter_2],
+            'protectedEntitiesPIAP': [protected_entities_iter_1, protected_entities_iter_2],
+            'sasHarness': [sas_harness_0, sas_harness_1],
+            'dpConfig': {'cert': 'dp_client.cert', 'key': 'dp_client.key'},
+            'K': 2,
```

Comment:
The value of K should be implicit in the configuration if you follow the structure I suggested previously. This will reduce the opportunity for self-inconsistent configs (in fact, the current config contradicts itself as you earlier assumed that you will have at least 3 iterations).

Reply:
Updated   , Removed 'K'